test/system/src/test/java/io/pravega/test/system/SingleSubscriberUpdateRetentionStreamCutTest.java

```diff
@@ -0,0 +1,209 @@
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(SCOPE, clientConfig);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write a single event.
+        log.info("Writing event e1 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent("e1", SIZE_30_EVENT).join();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        readerGroupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(READER_GROUP);
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(READER_GROUP + "-" + 1,
+                READER_GROUP, new JavaSerializer<>(), readerConfig);
+
+        // Read one event.
+        log.info("Reading event e1 from {}/{}", SCOPE, STREAM);
+        EventRead<String> read = reader.readNextEvent(READ_TIMEOUT);
+        assertFalse(read.isCheckpoint());
+        assertEquals("data of size 30", read.getEvent());
+
+        // Update the retention stream-cut.
+        log.info("{} generating stream-cuts for {}/{}", READER_GROUP, SCOPE, STREAM);
+        CompletableFuture<Map<Stream, StreamCut>> futureCuts = readerGroup.generateStreamCuts(streamCutExecutor);
+        // Wait for 5 seconds to force reader group state update. This will allow for the silent
+        // checkpoint event generated as part of generateStreamCuts to be picked and processed.
+        Exceptions.handleInterrupted(() -> TimeUnit.SECONDS.sleep(5));
+        EventRead<String> emptyEvent = reader.readNextEvent(READ_TIMEOUT);
+        assertTrue("Stream-cut generation did not complete", Futures.await(futureCuts, 10_000));
+
+        Map<Stream, StreamCut> streamCuts = futureCuts.join();
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP, streamCuts);
+        readerGroup.updateRetentionStreamCut(streamCuts);
+
+        // Write two more events.
+        log.info("Writing event e2 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent("e2", SIZE_30_EVENT).join();
+        log.info("Writing event e3 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent("e3", SIZE_30_EVENT).join();
+
+        // Check to make sure truncation happened after the first event.
+        // The timeout is 5 minutes as the retention period is set to 2 minutes. We allow for 2 cycles to fully complete
+        // and a little longer in order to confirm that the retention has taken place.
+        AssertExtensions.assertEventuallyEquals(true, () -> controller.getSegmentsAtTime(
```

Comment:
I think that this is checking every few milliseconds by default. Maybe you can pick a less aggressive evaluation interval, given that this is going to take at least a minute.

Reply:
Similarly for this timeout.

Reply:
I've changed the check interval to 1 min.

Reply:
Well, if this is going to take 1 minute, what can happen is that you evaluate one second before the condition evaluates to true, and then the test waits for 1 minute doing nothing. Evaluating every 1 or 5 seconds should be fine.

Reply:
I've changed it to 1 sec